### PR TITLE
Add some missing endpoints

### PR DESCRIPTION
**Pull request by jllopes**

### CODE REVIEW

File: src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala [hunk 1]
```diff
@@ -8,7 +8,11 @@ import play.api.libs.json._
 case class Repository(name: String, full_name: String, description: String, scm: String,
                       created_on: LocalDateTime, updated_on: LocalDateTime, owner: String, size: Long,
                       has_issues: Boolean, is_private: Boolean, language: String,
-                      url: Seq[RepositoryUrl])
+                      url: Seq[RepositoryUrl]) {
```
Comment (bmbferreira) [inline src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala]: maybe we should use the plural form here, since it is a list of urls.

Comment (jllopes) [reply src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala]: Done

File: src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala [hunk 2]
```diff
@@ -8,7 +8,11 @@ import play.api.libs.json._
 case class Repository(name: String, full_name: String, description: String, scm: String,
                       created_on: LocalDateTime, updated_on: LocalDateTime, owner: String, size: Long,
                       has_issues: Boolean, is_private: Boolean, language: String,
-                      url: Seq[RepositoryUrl])
+                      urls: Seq[RepositoryUrl]) {
+
+  val sshUrl: String = s"ssh://$scm@bitbucket.org/$owner/$name"
```
[the remark this answers is not in the document]
Comment (jllopes) [reply src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala]: You're right, it is not, removed.

File: src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala [hunk 3]
```diff
@@ -8,7 +8,8 @@ import play.api.libs.json._
 case class Repository(name: String, full_name: String, description: String, scm: String,
                       created_on: LocalDateTime, updated_on: LocalDateTime, owner: String, size: Long,
                       has_issues: Boolean, is_private: Boolean, language: String,
-                      url: Seq[RepositoryUrl])
+                      urls: Seq[RepositoryUrl]) {
+}
```
Comment (bmbferreira) [inline src/main/scala/com/codacy/client/bitbucket/v2/Repository.scala]: you can also remove the brackets 👍

File: src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala [hunk 4]
```diff
@@ -68,19 +73,22 @@ class PullRequestServices(client: BitbucketClient) {
     client.postJson(Request(url, classOf[JsObject]), JsNull)
   }
 
-  def createComment(author: String, repo: String, prId: Int, body: String,
+  def createLineComment(author: String, repo: String, prId: Int, body: String,
```
Comment (xplosunn) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala]: If you want to specialise this method to be a line comment, the "file" and "lineTo" have to stop being optional.

File: src/test/scala/com/codacy/client/bitbucket/v2/UserSpecs.scala [hunk 5]
```diff
@@ -50,4 +50,97 @@ class UserSpecs extends FlatSpec with Matchers {
...
+                  |"html": {
+                  |"href": "https://bitbucket.org/testteam2/"
+                  |},
+                  |"avatar": {
+                  |"href": "https://bitbucket.org/account/testteam2/avatar/"
+                  |}
+                  |}
+                  |}
+                  |}
+                  |]
```
Comment (xplosunn) [inline src/test/scala/com/codacy/client/bitbucket/v2/UserSpecs.scala]: If you could have this json properly indented that would be great.

File: src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala [hunk 6]
```diff
@@ -1,12 +1,12 @@
...
 
-import com.codacy.client.bitbucket.v2.User
+import com.codacy.client.bitbucket.v2.Team
 import com.codacy.client.bitbucket.client.{BitbucketClient, Request, RequestResponse}
 
 class TeamServices(client: BitbucketClient) {
 
-  def list: RequestResponse[Seq[User]] = {
-    client.executePaginated(Request(s"https://bitbucket.org/api/2.0/teams", classOf[Seq[User]]))
+  def list(role: String): RequestResponse[Seq[Team]] = {
```
Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/TeamServices.scala]: Can we put some kind of default here?

File: src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala (hunk 4 above)
Comment (rtfpessoa) [inline src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala]: I think commit comment would be better

Comment (jllopes) [reply src/main/scala/com/codacy/client/bitbucket/v2/service/PullRequestServices.scala]: This happens in package v1 as well, should I also change it there?

**Pull request merged**